### DISCUSSION TITLE
Equivalent to "Reformat on file save"

### DISCUSSION THREAD

**Discussion by andrew-selvia** [scalameta/metals#5700] [Q&A]
Coming from IntelliJ, I'm accustomed to the **Reformat on file save** setting which I use to [automatically format my code via Scalafmt](https://scalameta.org/scalafmt/docs/installation.html#format-on-save). Is there an equivalent in the Metals/VS Code ecosystem?

**Comment by andrew-selvia** [comment]
I'm currently approximating via this task I run via `sbtn`:
```
~scalafmtAll
```
but it's a bit slower since it formats the whole project.

**Comment by kasiaMarek** [comment]
Metals uses scalafmt for formatting. There is an option in VSCode in `Formatting` `Editor: Format On Save`. This will automatically send a request to metals to perform formatting on the current file. So this should do exactly what you want.
